src/pocketmine/level/Level.php

```diff
@@ -936,10 +936,8 @@ public function checkSleep(){
 		}
 
 		if($resetTime){
-			$time = $this->getTime() % Level::TIME_FULL;
-
-			if($time >= Level::TIME_NIGHT and $time < Level::TIME_SUNRISE){
-				$this->setTime($this->getTime() + Level::TIME_FULL - $time);
+			if($this->getTimeOfDay() >= Level::TIME_NIGHT and $this->getTimeOfDay() < Level::TIME_SUNRISE){
+				$this->setTime($this->getTime() + Level::TIME_FULL - $this->getTimeOfDay());
```

Comment:
it will be better to keep the change minimal here, replace the original `$time` variable assignment. It requires three less calls.